src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +28,34 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (obj.getClass().equals(Rectangle.class)) {
+            Rectangle rectangle = (Rectangle) obj;
+            return (color != null && color.equals(rectangle.color)
+                    || (rectangle.color == null && color == null))
```

Comment:
```suggestion
                    || rectangle.color == color)
```
just a note: this can be simplified

Reply:
Sure. Done.